src/main/java/games/strategy/engine/lobby/server/userDB/DbUser.java

```diff
@@ -1,42 +1,98 @@
 package games.strategy.engine.lobby.server.userDB;
 
-import java.util.Date;
+import com.google.common.base.Preconditions;
+import games.strategy.engine.framework.startup.ui.InGameLobbyWatcher;
+import games.strategy.util.Util;
 
+/**
+ * Value object representing data needed to create a user in database.
+ */
 public class DbUser {
-  private final String name;
-  private final String email;
-  private final boolean isAdmin;
-  private final Date lastLogin;
-  private final Date joined;
+  public final String name;
+  public final String email;
+  public final boolean admin;
+
+  public enum Role {
+    NOT_ADMIN, ADMIN
+  }
+
+  public static class UserName {
+    public final String value;
+    public UserName(String name) {
+      this.value = name;
+    }
+  }
+  public static class UserEmail {
+    public final String value;
+    public UserEmail(String email) {
+      this.value = email;
+    }
+  }
+
+  /**
+   *  Convenience constructor for non-admin users.
+   */
+  public DbUser(UserName name, UserEmail email) {
+    this(name, email, Role.NOT_ADMIN);
+  }
+
 
   /**
-   * All-arg value object constructor.
+   * An all-args constructor.
    */
-  public DbUser(final String name, final String email, final boolean isAdmin, final Date lastLogin, final Date joined) {
-    this.name = name;
-    this.email = email;
-    this.isAdmin = isAdmin;
-    this.lastLogin = lastLogin;
-    this.joined = joined;
+  public DbUser(UserName name, UserEmail email, Role role) {
+    this.name = name.value;
+    this.email = email.value;
+    this.admin = (role == Role.ADMIN);
   }
 
-  public String getEmail() {
-    return email;
+  public boolean isValid() {
+    return getValidationErrorMessage() == null;
   }
+  /**
+   * @return if this user is valid.
+   */
+  // TODO: test me
```

Comment:
This method has 100% instruction coverage and 4 of 6 branches covered on line 57.  Not sure if the TODO is there to remind you to cover those remaining branches.  Otherwise, it can probably be removed.